cmd/abapEnvironmentPushATCSystemConfig.go

```diff
@@ -0,0 +1,235 @@
+
+	subOptions := convertATCSysOptions(config)
+
+	// Determine the host, user and password, either via the input parameters or via a cloud foundry service key
+	connectionDetails, err := autils.GetAbapCommunicationArrangementInfo(subOptions, "/sap/opu/odata4/sap/satc_ci_cf_api/srvd_a2x/sap/satc_ci_cf_sv_api/0001")
+	if err != nil {
+		return errors.Wrap(err, "Parameters for the ABAP Connection not available")
+	}
+
+	cookieJar, err := cookiejar.New(nil)
+	if err != nil {
+		return errors.Wrap(err, "Could not create a Cookie Jar")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		MaxRequestDuration: 180 * time.Second,
+		CookieJar:          cookieJar,
+		Username:           connectionDetails.User,
+		Password:           connectionDetails.Password,
+	}
+	client.SetOptions(clientOptions)
+
+	return pushATCSystemConfig(config, connectionDetails, client)
+
+}
+
+func pushATCSystemConfig(config *abapEnvironmentPushATCSystemConfigOptions, connectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) error {
+
+	filelocation, err := filepath.Glob(config.AtcSystemConfigFilePath)
+	//check ATC system configuration json
+	var atcSystemConfiguartionJsonFile []byte
+	if err == nil {
+		filename, err := filepath.Abs(filelocation[0])
+		if err == nil {
+			atcSystemConfiguartionJsonFile, err = ioutil.ReadFile(filename)
+		}
+	}
+	if err == nil {
+		err = handlePushConfiguration(config, atcSystemConfiguartionJsonFile, connectionDetails, client)
+	}
+	if err != nil {
+		return fmt.Errorf("Pushing ATC System Configuration failed: %w", err)
+	}
+	return nil
+}
+
+func handlePushConfiguration(config *abapEnvironmentPushATCSystemConfigOptions, atcSystemConfiguartionJsonFile []byte, connectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) error {
+	uriConnectionDetails := connectionDetails
+	uriConnectionDetails.URL = ""
+	connectionDetails.XCsrfToken = "fetch"
+
+	// Loging into the ABAP System - getting the x-csrf-token and cookies
+	resp, err := abaputils.GetHTTPResponse("HEAD", connectionDetails, nil, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, "Authentication on the ABAP system failed", connectionDetails)
+		return err
+	}
+	defer resp.Body.Close()
+
+	log.Entry().WithField("StatusCode", resp.Status).WithField("ABAP Endpoint", connectionDetails.URL).Debug("Authentication on the ABAP system successful")
+	uriConnectionDetails.XCsrfToken = resp.Header.Get("X-Csrf-Token")
+	connectionDetails.XCsrfToken = uriConnectionDetails.XCsrfToken
+
+	abapEndpoint := connectionDetails.URL
+	connectionDetails.URL = abapEndpoint + "/configuration"
+
+	jsonBody := atcSystemConfiguartionJsonFile
+	resp, err = abaputils.GetHTTPResponse("POST", connectionDetails, jsonBody, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, "Could not push the given ATC System Configuration from File: "+config.AtcSystemConfigFilePath, uriConnectionDetails)
```

Comment:
As discussed: you will probably replace the "HandleHTTPError" message. Please be aware that the response can be nil (in case of a timeout).